team_test.go

```diff
@@ -0,0 +1,199 @@
+package pagerduty
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"text/template"
+)
+
+func userID(offset, index int) int {
+	return offset + index
+}
+
+func lastIndex(length, index int) bool {
+	return length-1 == index
+}
+
+const membersResponseTemplate = `
+{
+    {{template "pageInfo" . }},
+    "members": [
+        {{- $length := len .roles -}}
+        {{- $offset := .offset -}}
+        {{- range $index, $role := .roles -}}
+        {
+            "user": {
+                "id": "ID{{userID $offset $index}}"
+            },
+            "role": "{{ $role }}"
+        }
+        {{- if not (lastIndex $length $index) }},
+        {{end -}}
+        {{- end }}
+    ]
+}
+`
+
+var memberPageTemplate = template.Must(pageTemplate.New("json").
+	Funcs(templateUtilityFuncs).
+	Parse(membersResponseTemplate))
+
+const (
+	testValidTeamID = "MYTEAM"
+	testAPIKey      = "MYKEY"
+	testBadURL      = "A-FAKE-URL"
+	testMaxPageSize = 3
+)
+
+var templateUtilityFuncs = template.FuncMap{
+	"lastIndex": lastIndex,
+	"userID":    userID,
+}
+
+var pageTemplate = template.Must(template.New("pageInfo").Parse(`
+    "more": {{- .more -}},
+    "limit": {{- .limit -}},
+    "offset": {{- .offset -}}
+`))
+
+type pageDetails struct {
+	lowNumber, highNumber, limit, offset int
+	more                                 bool
+}
+
+func genMembersRespPage(details pageDetails, t *testing.T) string {
+	if details.limit == 0 {
+		details.limit = 25 // Default to 25, PD's API default.
+	}
+
+	possibleRoles := []string{"manager", "responder", "observer"}
+	roles := make([]string, 0)
+	for ; details.lowNumber <= details.highNumber; details.lowNumber++ {
+		roles = append(roles, possibleRoles[details.lowNumber%len(possibleRoles)])
+	}
+
+	buffer := bytes.NewBufferString("")
+	err := memberPageTemplate.Execute(buffer, map[string]interface{}{
+		"roles":  roles,
+		"more":   details.more,
+		"limit":  details.limit,
+		"offset": details.offset,
+	})
+
+	if err != nil {
+		t.Fatalf("Failed to apply values to template: %v", err)
+	}
+
+	return string(buffer.String())
+}
+
+func genRespPages(amount,
+	maxPageSize int,
+	pageGenerator func(pageDetails, *testing.T) string,
+	t *testing.T) []string {
```

Comment:
```suggestion
	t *testing.T) []string {
	
```

I suggest adding a newline here because otherwise the type signature blends into the start of the body. Took me a minute to realize what was happening